test/parallel/test-fs-write-stream-patch-open.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+const common = require('../common');
+const fs = require('fs');
+
+const tmpdir = require('../common/tmpdir');
+tmpdir.refresh();
```

Comment:
If I read this correctly, `createWriteStream` will open the file and then `open` opens it again. Since the `fs.open` is done asynchronously, both complete and we end up with two open file descriptors for the same file.

To me, this looks like another good reason to go forward with this PR and deprecate `open`.

A simple solution would be to add something like `s.on('open', (fd) => fs.closeSync(fd));` right after `createWriteStream` below. But this looks brittle to me, I think it's easy to change the `fs` code in a way that makes this test start failing, for example if the code is changed to only emit `'open'` only once.

I think a better solution is to ignore the problem, since this is getting deprecated and there is no good and recommended way of closing both file descriptors. This can be accomplished running the test in a child process:

```suggestion

// Run in a child process because 'out' is opened twice, blocking the tmpdir
// and preventing cleanup.
if (process.argv[2] !== 'child') {
  // Parent
  const assert = require('assert');
  const { fork } = require('child_process');
  tmpdir.refresh();

  // Run test
  const child = fork(__filename, ['child'], { stdio: 'inherit' });
  child.on('exit', common.mustCall(function(code) {
    assert.strictEqual(code, 0);
  }));

  return;
}

// Child
```